plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/HiveS3Module.java

```diff
@@ -61,13 +65,28 @@ protected void setup(Binder binder)
 
     private void bindSecurityMapping(Binder binder)
     {
-        if (buildConfigObject(S3SecurityMappingConfig.class).getConfigFile().isPresent()) {
-            checkArgument(!buildConfigObject(HiveConfig.class).isS3SelectPushdownEnabled(), "S3 security mapping is not compatible with S3 Select pushdown");
-            checkArgument(!buildConfigObject(RubixEnabledConfig.class).isCacheEnabled(), "S3 security mapping is not compatible with Hive caching");
+        S3SecurityMappingConfig configuration = buildConfigObject(S3SecurityMappingConfig.class);
+        if (configuration.getConfigFilePath().isEmpty()) {
+            return;
+        }
 
-            newSetBinder(binder, DynamicConfigurationProvider.class).addBinding()
-                    .to(S3SecurityMappingConfigurationProvider.class).in(Scopes.SINGLETON);
+        if (isHttp(configuration)) {
+            binder.bind(S3SecurityMappingsProvider.class).to(UriBasedS3SecurityMappingsProvider.class).in(Scopes.SINGLETON);
+            httpClientBinder(binder).bindHttpClient("s3SecurityMapping", ForS3SecurityMapping.class)
+                    .withConfigDefaults(config -> config
+                            .setRequestTimeout(Duration.succinctDuration(10, TimeUnit.SECONDS))
+                            .setSelectorCount(1)
+                            .setMinThreads(1));
```

Comment:
Why these are set to `1`?

Reply:
Taken from @electrum 's suggestion in https://github.com/trinodb/trino/pull/6210#discussion_r620687229 
If you have any other suggestions I can update these values accordingly or just use the defaults.